### ISSUE TITLE
Deleted user: history lost

### ISSUE THREAD

**Issue by woodpeck** [openstreetmap/openstreetmap-website#1311]
Deleted users used to be changed to something like "user_12345" (number chosen for illustration only), and you were still able to e.g. view the edit history of such a user. This is an important feature for the community to understand what someone has done (imagine e.g. a vandal making 20 edits then asking for his account to be deleted - you still want to be able to identify which edits they made).

However, for reasons I failed to identify in the code, deleted users now show up with the user name "deleted" (e.g. http://www.openstreetmap.org/changeset/37413625), and no hyperlink.

This is definitely a recent change but I cannot pinpoint where and when exactly it was introduced.

Requesting the raw XML for such an object still reveals the "user_12345" name, but then requesting that user's history through a link like http://www.openstreetmap.org/user/user_12345/history yields an error message as well. It would be desirable to be able to access the user's history in this way even though the account was deleted.

This might not be such a recent change but in March this year it was definitely still possible to access deleted users' histories in this way.

**Comment by tomhughes** [comment]
Deleted users still are changed to that, they just aren't visible. As far as I know deleted users have never been visible, or at least not for a very long time.

**Comment by tomhughes** [comment]
Oh the link may have been hidden in some specific contexts sure, but that was mostly because the link didn't work anyway, and just the name of it effectively revealed the identity because the uid to name mapping is known.

**Comment by tomhughes** [comment]
So 80c46f54d4930deacb8664ff44703b8c3c5d088e introduced the current message but even before that they were shown as anonymous, along with very old users that have never agreed to make their edits public.

**Comment by tomhughes** [comment]
So in fact it was the previous edit from the same merge (341f6cd9feed774f0abdb337f36272d5ecc92549) that made them anonymous. See #1154.

**Comment by woodpeck** [comment]
Hmm, appears I was wrong with "in March this year it was definitely still possible to access deleted users' histories". But I'm adamant that until very recently, I was able to see whether two edits had been made by the same deleted user or by different deleted users...

**Comment by tomhughes** [comment]
I don't see how - as far as I can the changeset list code has refused to give you a list for a non-active user for at least the last five years.

**Comment by woodpeck** [comment]
Yes I concede the "give you a list" bit but I *was* told that one edit was by "user_1234" and the other by "user_2345", whereas now both edits are by "deleted". Or am I imagining that as well?

**Comment by pnorman** [comment]
That was my recollection, but I haven't tested a great deal so can't be certain.

**Comment by tomhughes** [comment]
Yes that was possible until that change in February.

**Comment by mmd-osm** [comment]
> However, for reasons I failed to identify in the code, deleted users now show up with the user name "deleted" (e.g. http://www.openstreetmap.org/changeset/37413625), and no hyperlink.

There's in fact a hyperlink at the very bottom of the changeset page, shown as "user_836238" and pointing to http://www.openstreetmap.org/user/user_836238/history 

I guess that link shouldn't be shown in the first place, as it links directly to an error page "User user_836238 does not exist".

**Comment by BushmanK** [comment]
Currently, when you click "user_836238" on a page like http://www.openstreetmap.org/changeset/37413625 , it leads you to a page with some gibberish in the left sidebar. It actually loads the whole content of the http://www.openstreetmap.org/user/user_836238/history into the `div#sidebar_content`, which is obviously an improper behavior.

**Comment by Woazboat** [comment]
Being able to view the changeset history of a deleted user would be really nice to have. It's still possible via tools such as OsmCha, but having to rely on external tools here is suboptimal. I don't really see a reason why the changeset history of deleted users should be hidden.

As an example: User https://www.openstreetmap.org/user/D%20Hood (now https://www.openstreetmap.org/user/user_12421087) was recently caught vandalizing OSM by re-adding a no-longer existing golf course just so they could import it into their golf game: https://www.openstreetmap.org/changeset/132671033
They ended up leaving with the comment ["go ahead and delete it if you dont like it. I have the data I need."](https://www.openstreetmap.org/changeset/131929180) and deleted their own account. Because of that, it's no longer possible to check their recent edits for similar vandalism via https://www.openstreetmap.org/user/D%20Hood/history

**Comment by SomeoneElseOSM** [comment]
@Woazboat (separately from whether https://www.openstreetmap.org/user/user_12421087 should return anything useful) you can still get the user's changeset history from the API, just not through the "browse" pages, so if your goal here is "I want to download this user's history", you can, using e.g. https://github.com/woodpeck/osm-revert-scripts/blob/master/download_changesets_uid.sh .

**Comment by Woazboat** [comment]
Thank you, I know it's still possible via the API. I just think it should also be possible to simply view the changeset history on the website like for normal users.
